go/engine/login_provision.go

```diff
@@ -355,19 +359,25 @@ func (e *loginProvision) getValidPaperKey(ctx *Context) (*keypair, error) {
 		uid, err := e.uidByKID(kp.sigKey.GetKID())
 		if err != nil {
 			e.G().Log.Debug("getValidPaperKey attempt %d (%s): %s", i, prefix, err)
-			if nf, ok := err.(libkb.NotFoundError); ok {
-				// make Msg a little friendlier (instead of KID Not Found)
-				nf.Msg = ("paper key not found, most likely due to a typo in one of the words in the phrase")
-				lastErr = nf
-			} else {
+
+			switch err := err.(type) {
+			case libkb.NotFoundError:
+				lastErr = paperKeyNotFound
+			case libkb.AppStatusError:
+				if err.Code == libkb.SCNotFound {
+					lastErr = paperKeyNotFound
+				} else {
+					lastErr = err
+				}
+			default:
 				lastErr = err
 			}
+
 			continue
 		}
 
 		if uid.NotEqual(e.arg.User.GetUID()) {
-			e.G().Log.Debug("paper key (%s) entered was for a different user", prefix)
-			lastErr = fmt.Errorf("paper key (%s) valid, but for %s, not %s", prefix, uid, e.arg.User.GetUID())
```

Comment:
This is a horrible error to show to the user...

Reply:
can this error still make it to the debug logs?

Reply:
I thought it was bad to show it there too...but i can put it back.  Just seemed like if someone randomly entered someone else's paper key, we shouldn't advertise it.

Reply:
oh you're right. ok, forget it.